test/reducer_test.dart

```diff
@@ -2128,15 +2097,20 @@ main() {
 }
 ''';
   Design simple_helix_no_seq_design = Design.from_json(jsonDecode(simple_helix_no_seq_json));
-  test('Testing DNAEndsMoveStart', () {
+  test('Testing_DNAEndsMoveStart', () {
     AppState initial_state = app_state_from_design(simple_helix_no_seq_design);
     AppState actual_state = app_state_reducer(
         initial_state, DNAEndsMoveStart(offset: 0, helix: simple_helix_no_seq_design.helices[0]));
     AppState expect_state = initial_state.rebuild((b) => b.ui_state.dna_ends_are_moving = true);
-    expect_app_state_equal(actual_state, expect_state);
+
+    expect(actual_state.ui_state.changed_since_last_save, false);
+    expect(actual_state.ui_state.dna_ends_are_moving, true);
+    expect(actual_state.undo_redo.undo_stack, BuiltList<Design>());
```

Comment:
There's a lot of lines like this where the Undo and Redo stacks are being tested. It's not clear why. In this example, the test is named "Testing_DNAEndsMoveStart", which is describing a different behavior.

In general, each test should only be testing for certain functionality. The fact that the Undo and Redo stacks happen to be empty after this is an implementation detail of the fact that we started with an AppState that happened to have empty Undo and Redo stacks, and DNAEndsMoveStart is not a Design-changing Action.

This is just one example. In general, each unit test should be testing a particular piece of functionality. We might want a separate set of tests for Undo/Redo specifically, but outside of those, we shouldn't be checking the Undo/Redo stacks. Imagine if we change how they are represented. Then all these tests would have to be re-written, even though they aren't testing Undo/Redo behavior.